tests/utils/managed_process.py

```diff
@@ -154,48 +168,71 @@ def _start_process(self):
-            )
-
-            self._tee_proc = subprocess.Popen(
-                ["tee", self._log_path], stdin=self._sed_proc.stdout
-            )
+        # Single subprocess with unified output handling
+        self._proc = subprocess.Popen(
+            self.command,
+            env=self.env or os.environ.copy(),
+            cwd=self.working_dir,
+            stdin=subprocess.DEVNULL,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            start_new_session=True,  # Isolate process group to prevent kill 0 from affecting parent
+            text=True,  # Use text mode for easier line processing
+        )
 
-        else:
-            with open(self._log_path, "w", encoding="utf-8") as f:
-                self.proc = subprocess.Popen(
-                    self.command,
-                    env=self.env or os.environ.copy(),
-                    cwd=self.working_dir,
-                    stdin=stdin,
-                    stdout=stdout,
-                    stderr=stderr,
-                    start_new_session=True,  # Isolate process group to prevent kill 0 from affecting parent
-                )
-
-                self._sed_proc = subprocess.Popen(
-                    ["sed", "-u", f"s/^/[{self._command_name.upper()}] /"],
-                    stdin=self.proc.stdout,
-                    stdout=f,
-                )
-            self._tee_proc = None
+        # Single output processing function for both cases
+        def process_output():
+            try:
+                # Clear log file first
+                with open(self._log_path, "w", encoding="utf-8") as f:
+                    pass
+
+                # Process output line by line with non-blocking reads
+                while True:
+                    # Check if process is still alive
+                    if self._proc.poll() is not None:
+                        break
+
+                    # Use select to check if there's data available (non-blocking)
+                    try:
+                        ready, _, _ = select.select(
+                            [self._proc.stdout], [], [], 0.1
+                        )  # 100ms timeout
+                    except (OSError, ValueError):
+                        # File descriptor is closed or invalid
+                        break
+
+                    if ready:
+                        try:
+                            line = self._proc.stdout.readline()
+                            if not line:  # EOF reached
+                                break
+
+                            formatted_line = (
+                                f"[{self._command_name.upper()}] {line.rstrip()}"
+                            )
+                            if self.display_output:
+                                print(formatted_line, flush=True)
+                            # Write to log file
+                            with open(self._log_path, "a", encoding="utf-8") as f:
```

Comment:
nit: optimization would be to put this on the outside of the loop so you don't have to keep opening and closing the file? Not sure how python handles this.